addons/api/mirage/route-handlers/target.js

```diff
@@ -0,0 +1,109 @@
+export function targetHandler({ targets }, { params: { idMethod } }) {
+  const attrs = this.normalizedRequestAttrs();
+  const id = idMethod.split(':')[0];
+  const method = idMethod.split(':')[1];
+  const target = targets.find(id);
+  const {
+    attributes: {
+      brokeredCredentialSourceIds: originalBrokeredCredentials,
+      injectedApplicationCredentialSourceIds: originalInjectedCredentials,
+    },
+  } = target;
+  const selectedBrokeredCredentials = attrs.brokeredCredentialSourceIds;
+  const selectedInjectedCredentials =
+    attrs.injectedApplicationCredentialSourceIds;
+  const selectedHostSourceIds = attrs.hostSourceIds;
+  const originHostSetIds = target.hostSetIds;
+
+  const updatedAttrs = {
+    version: attrs.version,
+  };
+
+  const addToSourcesList = (originalSourcesIds, selectedSourcesIds) => {
+    const listOfSources = new Set(originalSourcesIds);
+    for (const elem of selectedSourcesIds) {
+      listOfSources.add(elem);
+    }
+    return Array.from(listOfSources);
+  };
+
+  const removeFromSourcesList = (originalSourceIds, selectedSourceIds) => {
+    const listOfSources = new Set(originalSourceIds);
+    for (const elem of selectedSourceIds) {
+      listOfSources.delete(elem);
+    }
+    return Array.from(listOfSources);
+  };
+
+  // If adding host sources, push them into the array
+  if (method === 'add-host-sources') {
+    if (selectedHostSourceIds) {
+      const listOfHostSources = addToSourcesList(
+        originHostSetIds,
+        selectedHostSourceIds
+      );
+      if (listOfHostSources?.length) {
+        updatedAttrs.hostSetIds = listOfHostSources;
+      }
+    }
+  }
+  // If deleting host sources, filter them out of the array
+  if (method === 'remove-host-sources') {
+    if (selectedHostSourceIds) {
+      const removedHostSources = removeFromSourcesList(
+        originHostSetIds,
+        selectedHostSourceIds
+      );
+      updatedAttrs.hostSetIds = removedHostSources;
+    }
+  }
+  // If adding brokered or injected application cred sources, push them into the array
+  if (method === 'add-credential-sources') {
+    if (selectedBrokeredCredentials) {
+      const listOfBrokeredCredentialSources = addToSourcesList(
+        originalBrokeredCredentials,
+        selectedBrokeredCredentials
+      );
+      if (listOfBrokeredCredentialSources?.length) {
+        updatedAttrs.brokeredCredentialSourceIds =
+          listOfBrokeredCredentialSources;
+        target.attributes.brokeredCredentialSourceIds =
+          listOfBrokeredCredentialSources;
```

Comment:
Why both?  The `target.update()` call below is using the `updatedAttrs`, which commits the values to the mock record.

Reply:
They are not nested attributes.

Reply:
It is not nested in the actual target model. But when mocking this target model, adding cred source ID fields directly to the target threw some errors in mirage as these fields are not actual models, hence I had it add those to target attributes. https://github.com/hashicorp/boundary-ui/pull/1269/files#diff-cbf1b6ba7d0d7fca2c5312169053b033bfc23b1663dc415b04e7dc30b60e2051L72

Reply:
The new cred source fields can't be expressed as relationships in the Mirage models the way they previously were.  As long as you omit the relationships in the Mirage model file, you should be able to treat them as regular fields without nesting them.

Reply:
actually I did try this and it didn't work because I somehow did not commit the code which removes the existing target model relationship! thanks Randy, it seems to work without the nested structure and I will update the pr

Reply:
done!